plugins/inputs/zipkin/codec/jsonV1/jsonV1.go

```diff
@@ -0,0 +1,252 @@
+package jsonV1
+
+import (
+	"encoding/json"
+	"fmt"
+	"strconv"
+	"time"
+
+	"github.com/influxdata/telegraf/plugins/inputs/zipkin/codec"
+	"github.com/openzipkin/zipkin-go-opentracing/_thrift/gen-go/zipkincore"
+)
+
+// JSON decodes spans from  bodies `POST`ed to the spans endpoint
+type JSON struct{}
+
+// Decode unmarshals and validates the JSON body
+func (j *JSON) Decode(octets []byte) ([]codec.Span, error) {
+	var spans []span
+	err := json.Unmarshal(octets, &spans)
+	if err != nil {
+		return nil, err
+	}
+
+	res := make([]codec.Span, len(spans))
+	for i := range spans {
+		if err := spans[i].Validate(); err != nil {
+			return nil, err
+		}
+		res[i] = &spans[i]
+	}
+	return res, nil
+}
+
+type span struct {
+	TraceID  string             `json:"traceId"`
+	SpanName string             `json:"name"`
+	ParentID string             `json:"parentId,omitempty"`
+	ID       string             `json:"id"`
+	Time     *int64             `json:"timestamp,omitempty"`
+	Dur      *int64             `json:"duration,omitempty"`
+	Debug    bool               `json:"debug,omitempty"`
+	Anno     []annotation       `json:"annotations"`
+	BAnno    []binaryAnnotation `json:"binaryAnnotations"`
+}
+
+func (s *span) Validate() error {
+	var err error
+	check := func(f func() (string, error)) {
+		if err != nil {
+			return
+		}
+		_, err = f()
+	}
+
+	check(s.Trace)
+	check(s.SpanID)
+	check(s.Parent)
+	if err != nil {
+		return err
+	}
+
+	_, err = s.BinaryAnnotations()
+	return err
+}
+
+func (s *span) Trace() (string, error) {
+	if s.TraceID == "" {
+		return "", fmt.Errorf("Trace ID cannot be null")
+	}
+	return TraceIDFromString(s.TraceID)
+}
+
+func (s *span) SpanID() (string, error) {
+	if s.ID == "" {
+		return "", fmt.Errorf("Span ID cannot be null")
+	}
+	return IDFromString(s.ID)
+}
+
+func (s *span) Parent() (string, error) {
+	if s.ParentID == "" {
+		return "", nil
+	}
+	return IDFromString(s.ParentID)
+}
+
+func (s *span) Name() string {
+	return s.SpanName
+}
+
+func (s *span) Annotations() []codec.Annotation {
+	res := make([]codec.Annotation, len(s.Anno))
+	for i := range s.Anno {
+		res[i] = &s.Anno[i]
+	}
+	return res
+}
+
+func (s *span) BinaryAnnotations() ([]codec.BinaryAnnotation, error) {
+	res := make([]codec.BinaryAnnotation, len(s.BAnno))
+	for i, a := range s.BAnno {
+		if a.Key() != "" && a.Value() == "" {
+			return nil, fmt.Errorf("No value for key %s at binaryAnnotations[%d]", a.K, i)
+		}
+		if a.Value() != "" && a.Key() == "" {
+			return nil, fmt.Errorf("No at binaryAnnotations[%d]", i)
```

Comment:
Not sure what this error means, also is binaryAnnotations going to be clear to user?